apis/python/src/tiledbsoma/util.py

```diff
@@ -127,3 +129,58 @@ def slice_to_range(ids: slice) -> Optional[Tuple[int, int]]:
     if ids.step is not None and ids.step != 1:
         raise ValueError("slice step must be 1 or None")
     return (ids.start, ids.stop)  # XXX TEMP
+
+
+def dense_indices_to_shape(
+    coords: Tuple[Union[int, slice], ...],
```

Comment:
coords should be typed with the DenseNdCoordinates type used by callers - that will help keep all of the various function signatures in sync (eg, would point out that dense coords can't take a pa.Array :)

Reply:
@bkmartinjr good call! Just know that since we use `typeguard` in our unit tests for #574, we'll no longer be able to write a unit test that passes in `pa.Array` and expect it to fail -- since `typeguard` will fail our _attempt_ to do that. :(